packages/@aws-cdk/aws-ec2/lib/subnet.ts

```diff
@@ -112,4 +126,53 @@ class ContainsIpAddressesSubnetFilter extends SubnetFilter {
       return cidrBlockObjs.some(cidr => subnetCidrBlock.containsCidr(cidr));
     });
   }
-}
\ No newline at end of file
+}
+
+/**
+ * Chooses subnets based on the subnetId
+ */
+class SubnetIdSubnetFilter extends SubnetFilter {
+
+  private readonly subnetIds: string[];
+
+  constructor(subnetIds: string[]) {
+    super();
+    this.subnetIds = subnetIds;
+  }
+
+  /**
+   * Executes the subnet filtering logic.
+   */
+  public selectSubnets(subnets: ISubnet[]): ISubnet[] {
+    return subnets.filter(subnet => this.subnetIds.includes(subnet.subnetId));
+  }
+}
+
+/**
+ * Chooses subnets based on the CIDR Netmask
+ */
+class CidrMaskSubnetFilter extends SubnetFilter {
+  private readonly mask: number
+
+  constructor(mask: number) {
+    super();
+    if (mask < 16 || mask > 28 ) {
+      throw new InvalidCidrRangeError(`x.x.x.x/${mask}`);
+    }
+    this.mask = mask;
+  }
+
+  /**
+   * Executes the subnet filtering logic.
+   */
+  public selectSubnets(subnets: ISubnet[]): ISubnet[] {
+    return this.retainByCidrMask(subnets, this.mask);
+  }
+
+  private retainByCidrMask(subnets: ISubnet[], netmask: number): ISubnet[] {
```

Comment:
`~` (Nit) I would probably just inline this; I don't know that making it its own method adds anything here.